src/SLCore/Analysis/SLCoreAnalyzer.cs

```diff
@@ -80,7 +80,7 @@ public void ExecuteAnalysis(string path, Guid analysisId, IEnumerable<AnalysisLa
             return;
         }
 
-        var extraProperties = GetExtraProperties(detectedLanguages);
+        var extraProperties = GetExtraProperties(path, detectedLanguages);
```

Comment:
Maybe this was already discussed somewhere. I am just bringing it up to raise awareness:

It seems before SLOOP, we used to have a background thread pool, and multiple analyses for multiple files are scheduled on parallel when disabling/enabling a rule:

https://github.com/SonarSource/sonarlint-visualstudio/blob/70d3c823af17f424a48abf73c5d067e229a1f272/src/CFamily/Analysis/CLangAnalyzer.cs#L122-L123

OTOH, after SLOOP, it seems that we schedule an analysis for each file separately, and these are handled sequentially by SLCORE later.

```
[ThreadId 94] [DEBUG] [SLCORE] Triggering analysis with configuration: [
  baseDir: C:\Users\user\source\repos\Project4\Project4
  extraProperties: {sonar.cfamily.compile-commands=C:\Users\user\AppData\Local\Temp\SLVS\VCXCD\0bc4e7e2-92cd-4fd7-bee0-07c8ec2d657b\FileName3.cpp.113293283.json}
  activeRules: [424 cpp, 189 c, 315 csharpsquid, 143 vbnet, 90 secrets]
  inputFiles: [
    file:///C:/Users/user/source/repos/Project4/Project4/FileName3.cpp (UTF-8)
  ]
]

[ThreadId 55] [DEBUG] [SLCORE] Start analysis
[ThreadId 46] [DEBUG] [SLCORE] Index files
[ThreadId 46] [DEBUG] [SLCORE] Language of file "file:///C:/Users/user/source/repos/Project4/Project4/FileName3.cpp" is detected to be "CPP"
[ThreadId 46] [DEBUG] [SLCORE] 1 file indexed
[ThreadId 16] [DEBUG] [SLCORE] Execute Sensor: CFamily
[ThreadId 5] [DEBUG] [SLCORE] Parsing compilation database C:\Users\user\AppData\Local\Temp\SLVS\VCXCD\0bc4e7e2-92cd-4fd7-bee0-07c8ec2d657b\FileName3.cpp.113293283.json
[ThreadId 5] [DEBUG] [SLCORE] Time to parse Compilation Database: 4ms
[ThreadId 5] [DEBUG] [SLCORE] Probing compiler: [C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.41.34120\bin\HostX64\x64\cl.exe]
[ThreadId 81] [DEBUG] [SLCORE] Git Repository not found for C:\Users\user\source\repos\Project4\Project4. The path C:\Users\user\source\repos\Project4\Project4 is not in a Git repository
[ThreadId 81] [DEBUG] [SLCORE] Triggering analysis with configuration: [
  baseDir: C:\Users\user\source\repos\Project4\Project4
  extraProperties: {sonar.cfamily.compile-commands=C:\Users\user\AppData\Local\Temp\SLVS\VCXCD\0bc4e7e2-92cd-4fd7-bee0-07c8ec2d657b\FileName2.cpp.-1362488898.json}
  activeRules: [424 cpp, 189 c, 315 csharpsquid, 143 vbnet, 90 secrets]
  inputFiles: [
    file:///C:/Users/user/source/repos/Project4/Project4/FileName2.cpp (UTF-8)
  ]
]

[ThreadId 46] [DEBUG] [SLCORE] Git Repository not found for C:\Users\user\source\repos\Project4\Project4. The path C:\Users\user\source\repos\Project4\Project4 is not in a Git repository
[ThreadId 46] [DEBUG] [SLCORE] Triggering analysis with configuration: [
  baseDir: C:\Users\user\source\repos\Project4\Project4
  extraProperties: {sonar.cfamily.compile-commands=C:\Users\user\AppData\Local\Temp\SLVS\VCXCD\0bc4e7e2-92cd-4fd7-bee0-07c8ec2d657b\FileName.cpp.-1470143858.json}
  activeRules: [424 cpp, 189 c, 315 csharpsquid, 143 vbnet, 90 secrets]
  inputFiles: [
    file:///C:/Users/user/source/repos/Project4/Project4/FileName.cpp (UTF-8)
  ]
]
```

Would it make sense to have an improvement ticket schedule analysis of all files in a single request?

For context, this can allow parallelization to happen on the analyzer side (to replace the old background thread pool), and is consistent with the behavior on CLion after changing the quality profile:

```
[Info - 11:49:52.871] Triggering analysis with configuration: [
  inputFiles: [
    file:///path/to/File1.cpp (UTF-8) [cpp]
    file:///path/to/File2.cpp (UTF-8) [cpp]
    file:///path/to/File3.cpp (UTF-8) [cpp]
  ]
]
```

Reply:
Good point, but not sure how to implement it on our side. I have posted about this on our slack. As this is not directly caused by this PR, I will resolve and we can continue the discussion later